File: Behavioral/Visitor/Java/RenderVisitor.java
// RenderVisitor.java

public class RenderVisitor implements ShapeVisitor {
  @Override
  public void visit(Circle circle) {
    System.out.println("Rendering a Circle.");
  }

  @Override
  public void visit(Square square) {
    System.out.println("Rendering a Square.");
  }
}

File: Behavioral/Visitor/Cpp/RenderVisitor.cpp
// RenderVisitor.cpp

#include "RenderVisitor.h"
#include "Circle.h"
#include "Square.h"

void RenderVisitor::visit(Circle& circle) {
  std::cout << "Rendering a Circle." << std::endl;
}

void RenderVisitor::visit(Square& square) {
  std::cout << "Rendering a Square." << std::endl;
}
